### PR TITLE
Checkstyle shadows vars pt9

### PR DESCRIPTION
**Pull request by pugnascotia**
Part of #19752. Fix more instances where local variable names were
shadowing field names.

### CODE REVIEW

File: x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/ExecuteStepsUpdateTaskTests.java [hunk 1]
```diff
@@ -140,17 +140,17 @@ private IndexMetadata setupIndexPolicy(String policyName) {
         lifecycleState.setPhase("new");
         lifecycleState.setAction("init");
         lifecycleState.setStep("init");
-        IndexMetadata indexMetadata = IndexMetadata.builder(indexName)
+        IndexMetadata _indexMetadata = IndexMetadata.builder(indexName)
```
Comment (mark-vieira) [inline x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/ExecuteStepsUpdateTaskTests.java:143]: We don't typically use underscore prefixed variable names, do we?

**Review by mark-vieira — APPROVED** [review]
One comment, otherwise LGTM.

File: x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/ExecuteStepsUpdateTaskTests.java (hunk 1 above)
Comment (pugnascotia) [reply x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/ExecuteStepsUpdateTaskTests.java]: We don't, no. I think I just ran out of inspiration here, but I went back and changed it 👍

**Comment by pugnascotia** [comment]
Backported to `8.0` in e4e47c4a376.

**Pull request merged**